### PR TITLE
Auth with deno

### PR DESCRIPTION
**Pull request by icode247**
#### Please update the following checklist as applicable. Otherwise, the maintainers may close your pull request.

- [x] You created a branch from `develop` branch.
- [x] You PR is raised on `develop` branch and not on`main`.
- [x] Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/main/GUIDELINES.md).
- [x] You have read the [Contribution Guidelines](https://github.com/LoginRadius/engineering-portal/blob/main/CONTRIBUTING.md) before creating this PR.
- [x] Checked for any duplicate blog entry.
- [x] Checked for all grammatical errors in the blog.
- [x] You have not published the blog content elsewhere on the internet on platforms like Medium, Dev.to, etc.
- [x] The content is not copy-pasted from elsewhere on the internet.

### CODE REVIEW

File: content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+
+![deno run screenshot](./aPk23wT.png)
+
+Great! The app is running on port:8080. You can now head to the postman to test your routes.
+
+### Signup Route
+
+![Sign up](./D2ORbZ0.png)
+
+![](./trYDtXn.png)
```
Comment (raghunath-r-a) [inline content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md:597]: @icode247 Seems like the image specified here is missing in the folder. Please cross-check.

**Review by raghunath-r-a — CHANGES_REQUESTED** [review]
Hi @icode247 

Please review the following comments and please make changes as necessary. 

Thank you,
@raghunath-r-a

File: content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
+
+author: "Ekekenta Odionyenfe Clinton"
+tags: ["JWT", "Deno", "Authentication"]
+description: ""
+---
+
+## Introduction
+It is essential to protect your APIs fron unauthorized access. And **JSON Web Token** (JWT) authentication allows you to secure your APIs so that an unauthorized person will not have access. 
+
+**Deno** is a well secure JavaScript/Typescript serverside runtime environment.
```
[the remark this answers is not in the document]
Comment (icode247) [reply content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md:13]: Deno was fully covered in the `What is Deno?` section. I suggest we remove short Deno explanation.

File: content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md (hunk 1 above)
Comment (icode247) [reply content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md:593]: I have added the missing image

File: content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+
+![Get Task By Id Route Access Granted](./ypIaObE.png)
+
+Feel free to test the other routes to see how things work.
+
+## Conclusion
+
+In this tutorial, you've built a todo CRUD API and protected the routes from unauthorized access using JSON Web Token (JWT). You have learnt how to create an Oak server in Deno, connect MongoDB, implement JWT authentication, as well as create and authenticate CRUD routes in a Deno application. 
+
+The code for this tutorial is available [here](https://github.com/icode247/auth-with-deno) on Github. Feel free to clone and extend features of the application.
```
[the remark this answers is not in the document]
Comment (icode247) [reply content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md:652]: I have a created a pull request for the sample code.

File: content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+    credential: {
+      username: 'dbUserName',
+      password: 'dbPassword',
+      mechanism: 'SCRAM-SHA-1',
+      db: "admin",
+    },
+ })
+
+ console.log("Database connected!");
+ const db = client.database("deno_auth");
```
Comment (VershaGupta) [inline content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md:183]: @icode247  Unable to connect to DB. Facing following error:
Uncaught (in promise) Error: MongoError: "Connection failed: MongoError: {\"ok\":0,\"errmsg\":\"Invalid database name: ''\",\"code\":73,\"codeName\":\"InvalidNamespace\"}"

But it works when I updated this code by following code::
```
import { MongoClient } from "https://deno.land/x/mongo@v0.30.0/mod.ts";

const dbString = `DB_STRING`;
const client = new MongoClient();
await client.connect(dbString);
const db = client.database("deno_auth");
export default db;
```

**Review by raghunath-r-a — APPROVED** [review]

**Pull request merged**